sailor/sap_iot/_common.py

```diff
@@ -0,0 +1,57 @@
+import string
+
+from cachetools.func import ttl_cache
+
+from ..utils.config import SailorConfig
+from ..utils.oauth_wrapper import get_oauth_client
+
+
+@ttl_cache(maxsize=8, ttl=600)
+def _request_extension_url(service):
```

Comment:
I think the function name should also carry the name "assetcentral" or "assetcntrl" because this function is branded for assetcentral.

Reply:
I rather expect this to become generic once we need access to some other schema. Since we don't know yet what exactly we need (in terms of replacement/formatting rules in particular) for some other schema, and don't have access to any other schema it seems difficult to implement that functionality properly now. I've added a comment in the last PR to explain this. (Alternatively we could make `_request_extension_url` completely generic already, but that makes the `ASSETCNTRL`-specifics somewhat less explicit, as they are still present in the `request_url` formatting rules. Thoughts?

Reply:
I prefer to name things exactly for what they are and refactor [make it generic] when the time comes. 

you could fake the "genericness" now by adding some parameter `schemaId` which does nothing really but just replace the url query parameter value. 
`_request_extension_url(schemaId, service_name)` 
and then call the function explicitly, e.g.: `_request_extension_url('ASSETCNTRL', 'upload')`. 
I like that, too.

Essentially the entire "common" module code is inherently geared towards assetcentral.
In the `request_(upload|aggregates)_url` functions, you would then also need to add that schemaId parameter or rename the functions to something like `request_assetcentral_upload_url` to be consistent.

Reply:
ok yes, I guess that is more proper :)
schema now explicitly specified (but with `ASSETCNTRL` as default value...)